test/functional/mempool_dump.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test mempool dump.
+
+By default, bitcoind will dump mempool on shutdown and
+then reload it on startup. Dump can also happen when requested via RPC.
+
+Test is as follows:
+
+  - start node0
+  - check no mempool file exist
+  - dump mempool and verify it then exists
+  - move parent directory to cause fail to dump, verify exception
+
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+import os
+
+class MempoolDumpTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+
+    def run_test(self):
+        self.log.debug("Verify mempool file doesn't exists")
+        regtestdir = os.path.join(self.options.tmpdir, 'node0', 'regtest')
+        mempooldat = os.path.join(regtestdir, 'mempool.dat')
+        assert not os.path.isfile(mempooldat)
+        self.log.debug("Dump mempool to disk")
+        assert self.nodes[0].dumpmempool()
+        assert os.path.isfile(mempooldat)
+        moveddir = os.path.join(self.options.tmpdir, 'node0', 'regtest_bk')
```

Comment:
What you might try instead is to copy the dumped mempool from node0 to node1 and check that it can be loaded. (The nodes are not connected, but share the same blockchain)

Anyway, I just noticed you are using a clean blockchain and dump the empty mempool. Might be better to populate it a bit.